network/logical_bridge_test.go

```diff
@@ -3,161 +3,220 @@
-
-	expectedLogicalBridge := &_go.LogicalBridge{} // Create your expected response
-	mockEvpnClient.On("CreateLogicalBridge", mock.Anything, mock.Anything, mock.Anything, mock.Anything, mock.Anything).
-		Return(expectedLogicalBridge, nil)
-
-	resultLogicalBridge, err := mockEvpnClient.CreateLogicalBridge(context.TODO(), "lb1", 1000, 1000, "10.10.10.10/16")
-
+// TestCreateLogicalBridge tests the CreateLogicalBridge method
+func TestCreateLogicalBridge(t *testing.T) {
+	wantIPPefix := &pc.IPPrefix{
+		Addr: &pc.IPAddress{
+			Af: pc.IpAf_IP_AF_INET,
+			V4OrV6: &pc.IPAddress_V4Addr{
+				V4Addr: 3232235776,
+			},
+		},
+		Len: 24,
+	}
+
+	wantRequest := &pb.CreateLogicalBridgeRequest{
+		LogicalBridgeId: "lb1",
+		LogicalBridge: &pb.LogicalBridge{
+			Spec: &pb.LogicalBridgeSpec{
+				VlanId:       100,
+				Vni:          proto.Uint32(500),
+				VtepIpPrefix: wantIPPefix,
+			},
+		},
+	}
+	wantResponse := wantRequest.LogicalBridge
+
+	mockClient := mocks.NewLogicalBridgeServiceClient(t)
+	mockClient.EXPECT().CreateLogicalBridge(mock.Anything, wantRequest).
+		Return(wantResponse, nil)
+
+	connClosed := false
+	mockConn := mocks.NewConnector(t)
+	mockConn.EXPECT().NewConn().Return(
+		&grpc.ClientConn{},
+		func() { connClosed = true },
+		nil,
+	)
+
+	c, _ := NewLogicalBridgeWithArgs(
+		mockConn,
+		func(grpc.ClientConnInterface) pb.LogicalBridgeServiceClient {
+			return mockClient
+		},
+	)
+
+	response, err := c.CreateLogicalBridge(
+		context.Background(),
+		"lb1", 100, 500, "192.168.1.0/24",
+	)
 	assert.NoError(t, err)
-	assert.Equal(t, expectedLogicalBridge, resultLogicalBridge)
-	mockEvpnClient.AssertExpectations(t)
+	assert.True(t, proto.Equal(response, wantResponse))
+	assert.Equal(t, true, connClosed)
 }
 
-func TestCreateLogicalBridgeError(t *testing.T) {
-	mockEvpnClient := &mocks.EvpnClient{}
-
-	expectedError := errors.New("mocked error")
-	mockEvpnClient.On("CreateLogicalBridge", mock.Anything, mock.Anything, mock.Anything, mock.Anything, mock.Anything).
-		Return(nil, expectedError)
-
-	resultLogicalBridge, err := mockEvpnClient.CreateLogicalBridge(context.TODO(), "lb1", 1000, 1000, "10.10.10.10/16")
-
-	assert.Error(t, err)
-	assert.Nil(t, resultLogicalBridge)
-	assert.Equal(t, expectedError, err)
-	mockEvpnClient.AssertExpectations(t)
-}
-
-func TestDeleteLogicalBridgeSuccess(t *testing.T) {
-	mockEvpnClient := &mocks.EvpnClient{}
-
-	expectedResponse := &emptypb.Empty{} // Create your expected response
-	mockEvpnClient.On("DeleteLogicalBridge", mock.Anything, mock.Anything, mock.Anything).
-		Return(expectedResponse, nil)
-
-	resultResponse, err := mockEvpnClient.DeleteLogicalBridge(context.TODO(), "lb1", true)
-
+// TestDeleteLogicalBridge tests the DeleteLogicalBridge method
+func TestDeleteLogicalBridge(t *testing.T) {
+	wantRequest := &pb.DeleteLogicalBridgeRequest{
+		Name:         "//network.opiproject.org/bridges/lb1",
+		AllowMissing: true,
+	}
+	wantResponse := &emptypb.Empty{}
```

Comment:
we don't really need to test expected response for `Empty` type. It is just an empty message, which cannot be used in any way